nms/app/views/domain-proxy/CbsdsTable.tsx

```diff
@@ -125,32 +125,46 @@ function CbsdsTable(props: WithAlert) {
           },
         )
       : [];
-  }, [ctx.state.cbsds]);
+  });
 
-  const getDataFn = useCallback(
-    (query: Query<CbsdRowType>) => {
-      ctx.setPaginationOptions({
-        page: query.page,
-        pageSize: query.pageSize,
-      });
+  const data: Array<CbsdRowType> = useMemo(() => {
+    return mapCbsdsToTableData(ctx.state.cbsds);
+  }, [ctx.state.cbsds]);
 
+  const getDataFn = useCallback((query: Query<CbsdRowType>) => {
+    if (query.page === ctx.state.page && query.totalCount === ctx.state.totalCount && query.pageSize === ctx.state.pageSize) {
       return Promise.resolve({
-        data,
+        data: mapCbsdsToTableData(ctx.state.cbsds),
         page: ctx.state.page,
         totalCount: ctx.state.totalCount,
       });
-    },
-    [ctx, data],
-  );
+    }
+
+    ctx.setPaginationOptions({
+      page: query.page,
+      pageSize: query.pageSize,
+    });
+
+    return ctx.refetch({
+      page: query.page,
+      pageSize: query.pageSize,
+    }).then((data: PaginatedCbsds) => {
+      const result = {
+        data: mapCbsdsToTableData(data.cbsds),
+        page: query.page,
+        totalCount: data.total_count,
+      };
+
+      return result;
+    });
+  }, [ctx.state.page, ctx.state.pageSize, ctx.state.totalCount]);
 
   const currentCbsd = useMemo(() => {
     return ctx.state.cbsds
       ? ctx.state.cbsds.find(({id}) => id === currentRow.id)
       : undefined;
   }, [ctx.state.cbsds, currentRow]);
 
-  if (ctx.state.isLoading) return <LoadingFiller />;
-
```

Comment:
Why don't we need the loading filler any more?

Reply:
[material-table](https://material-table.com/#/docs/features/remote-data) has it's own spinner, it can be seen in the docs example, and you can also see a gif in the PR description, the loader is there.

So I believe this extra spinner is not needed here.
Moreover, with this spinner the Table will be removed from the DOM on each page load. This causes the table to mount again, and make a request for the first page every time (this was part of the bug).

So basically the library was used in a wrong way, and this loader is not needed, I believe.

Reply:
I've now added this explanation to the commit message.